x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/CsvTestsDataLoader.java

```diff
@@ -359,10 +358,22 @@ private static void loadCsvData(
                     }
                 }
                 lineNumber++;
+                if (builder.length() > BULK_DATA_SIZE) {
```

Comment:
In ES terms, whenever bulk indexing is mentioned, the number of documents is the one that's used as a "chunking" mechanism. I'm not saying that using the size of the StringBuilder as a mechanism is wrong, only that it's unusual not to look at how many documents there are in each bulk request.

Reply:
The issue was that the new file I was adding had very long lines (large documents), so we would need an unreasonably small document count to get that file to load, which would be inefficient for all the other files. This way most files don't get chunked, and a few files get chunked into only 2 or 3 blocks, which is I think quite efficient.

Reply:
That's fair.